feat: add rules and skills for agents (#1720)
stellio-hub/stellio-context-broker: .claude/rules/architecture.md

(entity types and attributes covered), `operations`
- `Mode.INCLUSIVE` — remote data is merged on equal footing with local data
- `Mode.AUXILIARY` — remote data fills in only what the local broker does not have; processed last during merge
- `Operation` — controls which NGSI-LD operations the CSR participates in (e.g., `FEDERATION_OPS`, `RETRIEVE_OPS`)

### Distributed consumption (read/query federation)

- When handling a read or query request, fan out to all matching CSRs concurrently; never serialize remote calls
- Merge remote results with local results according to mode semantics: INCLUSIVE on equal footing, AUXILIARY as
  fallback only
- A CSR failure must never abort the local response — degrade gracefully and report the failure in an
  `NGSILD-Warning` response header

### Distributed provision (write federation)

- Write operations (create, update, delete) are forwarded only to CSRs whose `operations` cover the requested
  operation
- Partial failures (some CSRs unreachable or rejecting) must be reported in a `207 Multi-Status` response;
  do not silently drop failures

> - Never forward to an AUXILIARY-mode CSR unless the spec explicitly mandates it

Reviewer: ```suggestion
- Never forward a provision operation to a CSR with the AUXILIARY mode.
```

1. `ContextSourceRegistrationService` selects CSRs matching the request filters (`CSRFilters`)
2. `DistributedEntityConsumptionService` (or `DistributedTemporalEntityConsumptionService` for temporal endpoints) fans
out HTTP calls to matching CSR endpoints **in parallel** using Arrow fx `parMap`
3. Each remote call returns `Either<NGSILDWarning, CompactedEntityWithCSR>` — failures produce a warning, not a hard error
4. `ContextSourceUtils.mergeEntities` / `mergeEntitiesLists` merges local and remote results:
   - Local data takes precedence for conflicting attributes
   - AUXILIARY CSRs are sorted last and only contribute missing attributes
   - Merge uses `IorNel<NGSILDWarning, Result>` to carry both partial results and warnings simultaneously
5. Warnings from unreachable or invalid CSRs are surfaced in the `NGSILD-Warning` HTTP response header — the overall
request never fails solely because a CSR is unavailable

### Rules for CSR code

- All remote HTTP calls go through `ContextSourceHttpUtils`; do not open `WebClient` calls directly in services
- Merge logic lives in `ContextSourceUtils`; keep it pure (no I/O, no Spring beans)

> - Never propagate a CSR failure as a 5xx to the client — downgrade to a warning

Reviewer: True for consumption operation, not for provision

Author: reworded this whole section on CSR and federation
